addons/account_check_printing/models/account_payment.py

```diff
@@ -123,7 +123,7 @@ def print_checks(self):
             }
         else:
             self.filtered(lambda r: r.state == 'draft').post()
-            self.write({'state': 'sent'})
+            self.filtered(lambda r: r.state != 'cancel').write({'state': 'sent'})
```

Comment:
Maybe I'm wrong but is it not better to filter the cancelled payments at the beginning of the method?